src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableEnumerable.DebuggerProxy.cs

```diff
@@ -0,0 +1,47 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Collections.Immutable
+{
+    /// <summary>
+    /// Displays immutable enumerables in the debugger.
+    /// </summary>
+    /// <typeparam name="T">The element type of the enumerable.</typeparam>
+    /// <remarks>
+    /// This class should only be used with immutable enumerables, since it
+    /// caches the enumerable into an array for display in the debugger.
+    /// </remarks>
+    internal sealed class ImmutableEnumerableDebuggerProxy<T>
```

Comment:
@justinvp I want to keep the name `ImmutableEnumerableDebuggerProxy` since I want readers to know it's only for use by immutable enumerables. (I have also considered `ImmutableDebuggerProxy`, but that could be interpreted to mean that the proxy itself is immutable.) However, I will take your advice of removing the period. I am planning to replace the remaining proxies with something in Common that all mutable collections can use, so the filename inconsistencies should not be there for long.